### PR TITLE
Demo App: Venmo Feature Shell

### PR DESCRIPTION
**Pull request by sshropshire**
### Summary of changes

 - This PR contains UI that will host the demo of `VenmoClient` in the future

 ### Checklist

 ~- [ ] Added a changelog entry~

- @sshropshire

### CODE REVIEW

File: Demo/src/main/java/com/paypal/android/ui/features/FeaturesView.kt [hunk 1]
```diff
@@ -53,8 +58,8 @@ fun FeaturesView(
     LazyColumn(
         modifier = Modifier
             .background(MaterialTheme.colorScheme.surface)
-            .padding(horizontal = UIConstants.paddingMedium)
-            .fillMaxSize()
+            .fillMaxSize(),
+        contentPadding = PaddingValues(bottom = UIConstants.paddingMedium)
```
Comment (sshropshire) [inline Demo/src/main/java/com/paypal/android/ui/features/FeaturesView.kt:62]: Some clipping occurs on the bottom of the `FeaturesView` scroll view content area. The `contentPadding` update here ensures that all buttons are equally visible.

File: Demo/src/main/java/com/paypal/android/ui/features/FeaturesView.kt [hunk 2]
```diff
@@ -84,7 +95,13 @@ fun FeatureOptions(
...
     Card(
         shape = CardDefaults.elevatedShape,
-        elevation = CardDefaults.elevatedCardElevation()
+        elevation = CardDefaults.elevatedCardElevation(),
+        modifier = Modifier
+            .padding(
+                start = UIConstants.paddingMedium,
+                end = UIConstants.paddingMedium,
+                bottom = UIConstants.paddingSmall
+            )
```
Comment (sshropshire) [inline Demo/src/main/java/com/paypal/android/ui/features/FeaturesView.kt:104]: More layout tweaks to address a visual bug that occurs when the `FeaturesView` has enough content to become scrollable.

File: Demo/src/main/java/com/paypal/android/ui/features/FeaturesView.kt [hunk 3]
```diff
@@ -98,14 +115,20 @@ fun FeatureOptions(
...
 fun FeatureGroupHeader(text: String) {
-    Spacer(modifier = Modifier.size(UIConstants.paddingLarge))
-    Text(
-        text = text,
-        color = MaterialTheme.colorScheme.onSurface,
-        fontWeight = FontWeight.Bold,
-        style = MaterialTheme.typography.titleLarge,
-    )
-    Spacer(modifier = Modifier.size(UIConstants.paddingMedium))
+    Row(
```
Comment (sshropshire) [inline Demo/src/main/java/com/paypal/android/ui/features/FeaturesView.kt:118]: Moving away from `Spacer` views here. Padding does the same job without adding unnecessary Compose elements.

File: Demo/src/main/java/com/paypal/android/ui/venmocheckout/VenmoCheckoutUiState.kt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.paypal.android.ui.venmocheckout
+
+import com.paypal.android.uishared.state.ActionState
+
+data class VenmoCheckoutUiState(
+    // TODO: replace once EligibilityResult is implemented
+//    val checkEligibilityState: ActionState<EligibilityResult, Exception> = ActionState.Idle,
```
Comment (sshropshire) [inline Demo/src/main/java/com/paypal/android/ui/venmocheckout/VenmoCheckoutUiState.kt]: Placeholder for `EligibilityClient` logic.

**Review by martita67 — APPROVED** [review]

**Review by scannillo — APPROVED** [review]

File: Demo/src/main/java/com/paypal/android/ui/venmocheckout/VenmoCheckoutView.kt [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import com.paypal.android.utils.UIConstants
+
+@ExperimentalComposeUiApi
+@Composable
+fun VenmoCheckoutView(
+    viewModel: VenmoCheckoutViewModel = hiltViewModel(),
+) {
+    val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+    val scrollState = rememberScrollState()
+
```
Comment (jaxdesmarais) [inline Demo/src/main/java/com/paypal/android/ui/venmocheckout/VenmoCheckoutView.kt:28]:
```suggestion
```

**Review by jaxdesmarais — APPROVED** [review]

**Pull request merged**